src/main/java/io/openliberty/tools/intellij/util/LibertyGradleUtil.java

```diff
@@ -104,10 +106,41 @@ public static boolean validBuildGradle(PsiFile file) throws IOException {
                     Pattern pattern2 = Pattern.compile(regex2);
                     Matcher matcher2 = pattern2.matcher(sub);
                     while (matcher2.find()) {
-                        return true;
+                        String plugin = sub.substring(matcher2.start(), matcher2.end());
+                        boolean vaildContainerVersion = containerVersion(plugin);
+
+                        return (new BuildFile(true, vaildContainerVersion));
                     }
                 }
             }
+        return (new BuildFile(false, false));
+    }
+
+    /**
+     * Given the plugin object as a string, use a regex to
+     * get the version.
+     *
+     * @param plugin plugin object as a string
+     * @return true if liberty-gradle-plugin is compatible for dev mode with containers
+     */
+    private static boolean containerVersion(String plugin) {
+        // get the version from the plugin
+        String versionRegex = "(?<=:liberty-gradle-plugin:).*(?=\')";
```

Comment:
Just a note, looks like this is expecting the plugin to be specified in a format like `io.openliberty.tools:liberty-gradle-plugin:3.1-M2`.  However, there are other possible formats as well, including:

- `plugins` block according to the last example in the section https://github.com/OpenLiberty/ci.gradle#adding-the-plugin-to-the-build-script
- group, name, version strings according to https://docs.gradle.org/current/userguide/declaring_dependencies.html#sec:dependency-types, for example:
```
buildscript {
    repositories {
        mavenCentral()
    }
    dependencies {
        classpath group: 'io.openliberty.tools', name: 'liberty-gradle-plugin', version: '3.1-M2'
    }
}
```

Reply:
Opened issue #26 to investigate this further